### ISSUE TITLE
Enhance Intrinsic::fmuladd translation.

### ISSUE THREAD

**Issue by haonanya** [KhronosGroup/SPIRV-LLVM-Translator#2162]
-cl-mad-enable is off by default, however ReplaceLLVMFmulAddWithOpenCLMad  is true by default https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/include/LLVMSPIRVOpts.h#L244C8-L244C40, which seems not consistent. Intrinsic::fmuladd should be translated into mad when "less-precise-fpmad" = "true" which indicates -cl-mad-enable is specified.

**Comment by wenju-he** [comment]
fmuladd should be translated to fmuladd as it is. It is likely the lack of support of the translation in spirv spec so that it is translate to mad.

**Comment by AlexeySachkov** [comment]
> fmuladd should be translated to fmuladd as it is. It is likely the lack of support of the translation in spirv spec so that it is translate to mad.

That is correct. We don't have a direct equivalent of `fmuladd` in SPIR-V and therefore we have decided to go with `mad` instead.

I'm not entirely sure if we _only_ need to use `mad` when `-cl-mad-enable` is present, because for FULL profile `mad` is expected to have the same semantics as `fmuladd`. Therefore, the current approach _is_ valid for FULL profile. At first glance, it seems to me that if someone is targeting an embedded profile where `mad` could be less precise, then they should switch the value of the option.
